sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4818,6 +4818,14 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val CHAR_VARCHAR_EXTENDS_STRING = buildConf("spark.sql.charVarcharExtendsString")
```

Comment:
how is this different from `LEGACY_CHAR_VARCHAR_AS_STRING`?

Reply:
Yeah, I agree this naming is not perfect. Maybe we could make clear what the config main idea is `CHAR_VARCHAR_AS_PROPER_TYPES` as we will replace char/varchar with CharType and VarcharType in the underlying code instead of using StringType.